modules/javafx.controls/src/test/java/test/javafx/scene/control/TextFieldTest.java

```diff
@@ -284,6 +298,85 @@
         assertEquals("x", dummyTxtField.getText());
     }
 
-
-
+    private Toolkit tk;
+    private Scene scene;
+    private Stage stage;
+    private StackPane root;
+
+    /**
+     * test for JDK-8207774: ENTER must not be forwared if actionHandler
+     * consumed the action.
+     *
+     * Here we test that an accelerator is not triggered.
+     */
+    @Test
+    public void testEnterWithConsumingActionHandlerAccelerator() {
+        initStage();
+        root.getChildren().add(txtField);
+        txtField.addEventHandler(ActionEvent.ACTION, e ->
+        e.consume()
+                );
```

Comment:
The indentation is a little odd here: `e.consume()` should be indented and the closing `)` need not be.